### PR TITLE
ENH: Dont allow length=1 regressions/correlations.

### PR DESCRIPTION
**Pull request by ssanderson**
They're not meaningful, and they cause warnings from numpy.

Implemented in terms of a new preprocessor, `expect_bounded`, which
takes a tuple of `upper_bound` and `lower_bound`.

### CODE REVIEW

File: zipline/utils/input_validation.py [hunk 1]
```diff
@@ -503,6 +503,97 @@ def _expect_element(collection):
     return preprocess(**valmap(_expect_element, named))
 
 
+def expect_bounded(**named):
+    """
+    Preprocessing decorator that verifies inputs fall between upper and lower
+    bounds.
```
Comment (richafrank) [inline zipline/utils/input_validation.py]: Could you add a comment here about the bounds being inclusive or exclusive?

File: zipline/utils/input_validation.py [hunk 2]
```diff
@@ -503,6 +503,97 @@ def _expect_element(collection):
...
+                " argument '{name}', but got {bounds} instead.".format(
+                    name=name,
+                    bounds=bounds,
+                )
+            )
+
+    def _expect_bounded(bounds):
+        (lower, upper) = bounds
+        if lower is None:
+            should_fail = lambda value: value > upper
```
Comment (richafrank) [inline zipline/utils/input_validation.py]: 👍 for the name `should_fail` - much clearer than `pred`.

**Comment by jdricklefs** [comment]
Wow, this is a fancy fix. Thanks @ssanderson

File: zipline/utils/input_validation.py (hunk 1 above)
Comment (ssanderson) [inline zipline/utils/input_validation.py]: Updated.

**Comment by richafrank** [comment]
LGTM

**Comment by dmichalowicz** [comment]
Thanks for this Scott! Good catch on the length-1 tests

**Pull request merged**